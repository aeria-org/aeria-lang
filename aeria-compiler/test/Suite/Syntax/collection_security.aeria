collection CollectionSecutiryTest {
  security {
    func1 {
      logging {
        strategy "tenant"
      }
      rateLimiting {
        strategy "ip"
        scale 10
      }
    }
  }
}
