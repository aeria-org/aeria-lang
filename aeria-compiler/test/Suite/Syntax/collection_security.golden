{
  "kind": "Program",
  "collections": [
    {
      "security": [
        {
          "rateLimiting": {
            "strategy": "ip",
            "scale": 10,
            "kind": "SecurityRateLimiting"
          },
          "logging": {
            "strategy": "tenant",
            "kind": "SecurityLogging"
          },
          "kind": "SecurityItem",
          "functionName": {
            "value": "func1",
            "kind": "FunctionName"
          }
        }
      ],
      "name": {
        "value": "CollectionSecutiryTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
